src/wallet/bdb.cpp

```diff
@@ -887,7 +887,7 @@ bool BerkeleyBatch::HasKey(DataStream&& key)
 
 bool BerkeleyBatch::ErasePrefix(Span<const std::byte> prefix)
 {
-    if (!TxnBegin()) return false;
+    assert(activeTxn); // because this is a read-write operation, the current db config requires a txn context.
```

Comment:
In 0dba0b6de03c0c79bb09296b913729ec8eb75aca "wallet: bdb batch 'ErasePrefix', do not create txn internally"

Instead of `assert`ing, I would prefer if this did `if (!Assume(activeTxn)) return false` so we didn't cause nodes to crash if a transaction were missing for whatever reason. `Assume` should still catch this for developers.

Reply:
> Instead of `assert`ing, I would prefer if this did `if (!Assume(activeTxn)) return false` so we didn't cause nodes to crash if a transaction were missing for whatever reason. `Assume` should still catch this for developers.

That would be risky in terms of consistency. `BerkeleyBatch::ErasePrefix` traverses the entire db, reading entries one by one, erasing the ones that match the key prefix. If we don't execute this function within a txn context, it's possible that certain records are removed while others remain due to an internal failure during the procedure.

Currently, the only function that uses `ErasePrefix()` is `WalletBatch::EraseAddressData`, which is exclusively called from `CWallet::DelAddressBook`, which is modified in the first two commits (thats one of the reasons behind taking those commits from #26836).

Reply:
Sure, that's why I suggest using `Assume`, and `return false` if there is no `activeTxn`. The callers of this should already be checking the return value to know whether the records were actually erased from the database.

Reply:
> Sure, that's why I suggest using `Assume`, and `return false` if there is no `activeTxn`. The callers of this should already be checking the return value to know whether the records were actually erased from the database.

ah, I completely skipped the `return false` last night.. sorry. All good, pushed.